File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int lo = 0;
        int hi = nums.length -1;
        if(nums[lo] < nums[hi])
            return nums[lo];
        while(lo < hi){
            int mid = lo + (hi-lo)/2;
            if(nums[mid] >= nums[0]){
                lo = mid+1;
            }
            else{
                hi = mid;
            }  
        }
        return nums[lo];
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        lo = 0
        hi = len(nums)-1
        if nums[0] < nums[hi]:
            return nums[0]
        while lo < hi:
            mid = lo + (hi-lo)//2
            if nums[mid] >= nums[0]:
                lo = mid+1
            else:
                hi = mid
        return nums[lo]
            
        
        